src/cts/src/Graphics.cpp

```diff
@@ -43,20 +43,28 @@
 
 namespace cts {
 
-Graphics::Graphics(HTreeBuilder* h_tree_builder, Clock* clock)
-    : clock_(clock), h_tree_builder_(h_tree_builder), sink_clustering_(nullptr)
+void Graphics::initializeWithClock(HTreeBuilder* h_tree_builder, Clock& clock)
 {
+  clock_ = &clock;
+  h_tree_builder_ = h_tree_builder;
+  sink_clustering_ = nullptr;
```

Comment:
The CTS implementation wants to initialize the object, and the object is now constructed outside of the CTS implementation, so it needs to be an initialization method (that inherits from the abstract API) instead of a constructor.